src/browser/BrowserMonitoringService.ts

```diff
@@ -0,0 +1,78 @@
+import { datadogLogs } from '@datadog/browser-logs';
+
+import { Logger } from '../shared/Logger';
+import { ConsoleLogger } from '../shared/ConsoleLogger';
+
+const ddLogger = datadogLogs.logger;
+
+interface RemoteMonitoringServiceParams {
+  serviceName: string;
+  serviceVersion: string;
+  serviceEnv: string;
+  clientToken: string;
+}
+
+export class BrowserMonitoringService {
+  private logger: Logger;
+  private remoteMonitoringServiceParams:
+    | RemoteMonitoringServiceParams
+    | undefined;
+
+  constructor(remoteMonitoringServiceParams?: RemoteMonitoringServiceParams) {
+    const { serviceName, serviceVersion, serviceEnv, clientToken } =
+      remoteMonitoringServiceParams ?? {};
+
+    const isSomeRemoteMonitoringServiceParamMissing =
+      !serviceName || !serviceVersion || !serviceEnv || !clientToken;
+
+    if (!remoteMonitoringServiceParams) {
+      this.logger = new ConsoleLogger();
+    } else if (
+      remoteMonitoringServiceParams &&
+      isSomeRemoteMonitoringServiceParamMissing
+    ) {
+      this.logger = new ConsoleLogger();
+      this.logger.warn(
+        '[MonitoringService] can not initialize remote monitoring service, because some variables are missing. Initializing with console instead',
+      );
+      this.logger.warn(`
+        serviceName - ${serviceName}
+        serviceVersion - ${serviceVersion}
+        serviceEnv - ${serviceEnv}
+        clientToken - ${clientToken}
+      `);
+    } else {
+      this.logger = ddLogger;
+    }
+
+    this.remoteMonitoringServiceParams = remoteMonitoringServiceParams;
+  }
+
+  setupReportingNativeLogs() {
+    if (!this.remoteMonitoringServiceParams) {
+      this.reportError(
+        new Error('Can not setup remote monitoring without all needed params'),
+      );
+
+      return;
+    }
+
+    datadogLogs.init({
```

Comment:
maybe it would make sense init logs in the constructor 🤔

Reply:
I am not sure.
https://stackoverflow.com/questions/4414471/initialize-vs-constructor-method-proper-usage-on-object-creation - i found smth like this (for not JS lang) but it kind of explain what I think about it. 
Namely, to me this init() call can be strong side effect, and I am not sure that I expect constructor to call side effects. 
For example I may want to have several instances of an object if I provide consumer with constructor - in such case that implicit work in constructor will conflict, even though consumer probably don't expect that. 

For example in case I wanna have remote and local monitoring systems in the same codebase.